### PR TITLE
feat(zk): implement `RotateOmega()`

### PR DESCRIPTION
**Pull request by lightscale-luke**
- implement RotateOmega() on UnivariateEvaluationDomain
   * the function is used on LookupArgument and PermutationArgument.

### CODE REVIEW

File: tachyon/zk/plonk/argument_common/rotate_omega.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  Field point = value;
+  if (rotation.value() >= 0) {
+    point *= domain->group_gen().Pow(math::BigInt<1>(rotation.value()));
+  } else {
+    point *=
+        domain->group_gen_inv().Pow(math::BigInt<1>(-1 * rotation.value()));
+  }
+
+  return point;
+}
```
Comment (chokobole) [inline tachyon/zk/plonk/argument_common/rotate_omega.h:29]: Could you implement this on `Rotation`?

**Review by chokobole — APPROVED** [review]
LGTM

File: tachyon/zk/plonk/argument_common/rotate_omega.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+#include "tachyon/math/polynomials/univariate/univariate_evaluation_domain.h"
+#include "tachyon/zk/plonk/circuit/rotation.h"
+
+namespace tachyon::zk {
+
+// Multiplies a value by some power of |omega|, essentially rotating over the
+// domain.
+template <typename Field, size_t MaxDegree>
+Field RotateOmega(
```
Comment (dongchangYoo) [inline tachyon/zk/plonk/argument_common/rotate_omega.h:17]: duplicated code!

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Pull request merged**